test/common/odp_cunit_common.c

```diff
@@ -273,6 +273,40 @@ static int default_term_func(void)
 	return odp_cunit_print_inactive();
 }
 
+static void _cunit_test_setup_func(void)
+{
+	CU_AllTestsCompleteMessageHandler all_test_comp_handler;
+	CU_SuiteCompleteMessageHandler suite_comp_handler;
+	CU_pFailureRecord pfailrec;
+	CU_pSuite psuite;
+
+	if (!getenv("ODP_CUNIT_FAIL_IMMEDIATE"))
+		return;
+
+	if (CU_get_number_of_failure_records() == 0)
+		return;
+
+	/* User wants the suite to fail immediately once a test hits an error */
+	psuite = CU_get_current_suite();
+	pfailrec = CU_get_failure_list();
+
+	printf("Force aborting as a previous test failed\n");
+
+	/* Call the Cleanup functions before aborting */
+	psuite->pCleanupFunc();
+
+	suite_comp_handler = CU_get_suite_complete_handler();
+	if (suite_comp_handler)
+		suite_comp_handler(psuite, pfailrec);
+
+	all_test_comp_handler = CU_get_all_test_complete_handler();
+	if (all_test_comp_handler)
+		all_test_comp_handler(pfailrec);
+
+	/* Set some CUnit error so that it aborts */
+	CU_set_error(CUE_NOTEST);
```

Comment:
This is a CUnit internal function, which doesn't do anything else than prints an error message (a bit confusing one in this case) and exits. So, it could be better to just call `exit(EXIT_FAILURE)` instead.